cli/template/extras/src/server/api/trpc/base.ts

```diff
@@ -51,20 +51,22 @@ export const createTRPCContext = (_opts: CreateNextContextOptions) => {
  * errors on the backend.
  */
 import { initTRPC } from "@trpc/server";
+import { TRPCErrorShape } from "@trpc/server/rpc";
 import superjson from "superjson";
 import { ZodError } from "zod";
 
 const t = initTRPC.context<typeof createTRPCContext>().create({
   transformer: superjson,
   errorFormatter({ shape, error }) {
-    return {
-      ...shape,
-      data: {
-        ...shape.data,
-        zodError:
-          error.cause instanceof ZodError ? error.cause.flatten() : null,
-      },
-    };
+    const data: TRPCErrorShape["data"] = { ...shape.data };
+    if (error.cause) {
+      if (error.cause instanceof ZodError) {
+        data.zodError = error.cause.flatten();
+      } else {
+        data.cause = error.cause;
```

Comment:
This is a little scary - I don't think you wanna send every single error down to the client, they may contain information not intended for the public (there is also `onError` in the API handler you can use to send logs of the error, which is only accessible on the server.